receiver/fluentforwardreceiver/conversion_test.go

```diff
@@ -0,0 +1,200 @@
+// Copyright 2020 The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package fluentforwardreceiver
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/tinylib/msgp/msgp"
+
+	"go.opentelemetry.io/collector/internal/data"
+	otlpcommon "go.opentelemetry.io/collector/internal/data/opentelemetry-proto-gen/common/v1"
+	logsproto "go.opentelemetry.io/collector/internal/data/opentelemetry-proto-gen/logs/v1"
+	otlpresource "go.opentelemetry.io/collector/internal/data/opentelemetry-proto-gen/resource/v1"
```

Comment:
It is not advised to use otlp structs directly in component code or int component tests. OTLP structs may change over time (e.g. by using custom gogoproto data types). We provide wrappers for the purpose of encapsulating and isolating component code from such changes.
I highly advise to rewrite these tests not to use otlp structs at all. Work with pdata only.

Reply:
Ok, I added a test util to easily construct logs in a declarative manner using the pdata methods only.